### PR TITLE
SPCTR-3011 Fix: Separator - Resolved an edge-case issue in which the responsive alignment would not work as intended.

### PR DESCRIPTION
**Pull request by AbhijitNage123**
### Description

- https://brainstormforce.atlassian.net/browse/SPCTR-3011

### Screenshots

- https://share.getcloudapp.com/xQuEzzB1

### Types of changes

- - Bug fix (non-breaking change which fixes an issue)

### How has this been tested?

- Added to the card.

### Checklist:
- [x] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: includes/blocks/separator/frontend.css.php [hunk 1]
```diff
@@ -32,12 +32,20 @@
...
+			'margin-right' => 'auto',
+			'margin-left'  => 'auto',
 		);
 	}
 	if ( 'right' === $attr['separatorAlign'] ) {
 		$align_css = array(
 			'margin-right' => 0,
+			'margin-left'  => 'auto',
 		);
 	}
```
Comment (bsf-zanev) [inline includes/blocks/separator/frontend.css.php:50]: Should we do this in a switch-case instead?

Comment (bsf-zanev) [reply includes/blocks/separator/frontend.css.php:50]: Same for responsive and in JS

File: src/blocks/separator/styling.js [hunk 2]
```diff
@@ -5,7 +5,30 @@
...
+			break;
+		case 'right':
+				alignCSS = {
+					'margin-right': 0,
+					'margin-left': 'auto',
+				};
+				break;
+	}
+	return alignCSS;
+}
```
Comment (bsf-zanev) [inline src/blocks/separator/styling.js:31]: Can we make this an arrow function? ( Just a preference )

Also it looks like the indentation inside the cases is 1 tab space off.

File: includes/blocks/separator/frontend.css.php [hunk 3]
```diff
@@ -14,6 +14,38 @@
...
+					'margin-left'  => 'auto',
+					'margin-right' => 'auto',
+				);
+			break;
+		case 'right':
+				$align_css = array(
+					'margin-right' => 0,
+					'margin-left'  => 'auto',
+				);
+			break;
```
Comment (bsf-zanev) [inline includes/blocks/separator/frontend.css.php:44]: Indentation seems a bit off in these ones too.

**Review by bsf-zanev — COMMENTED** [review]
The new code looks good, but just added a few practices here.
Do resolve conflicts as well.

File: includes/blocks/separator/frontend.css.php (hunk 3 above)
Comment (AbhijitNage123) [reply includes/blocks/separator/frontend.css.php]: Done

File: src/blocks/separator/styling.js (hunk 2 above)
Comment (AbhijitNage123) [reply src/blocks/separator/styling.js]: Done

File: includes/blocks/separator/frontend.css.php (hunk 3 above)
Comment (AbhijitNage123) [reply includes/blocks/separator/frontend.css.php]: PHPCS giving an error. Keeping it same

**Review by aapatre — APPROVED** [review]

**Review by bsf-zanev — APPROVED** [review]

File: src/blocks/separator/styling.js [hunk 4]
```diff
@@ -5,6 +5,30 @@
...
+						'margin-right': 'auto',
+					};
+					break;
+		case 'right':
+					alignCSS = {
+						'margin-right': 0,
+						'margin-left': 'auto',
+					};
+					break;
+	}
```
Comment (bsf-zanev) [inline src/blocks/separator/styling.js:29]: @AbhijitNage123 - Should we indent it like this?
( Same for PHP )

```suggestion
	switch ( align ) {
		case 'left':
			alignCSS = {
				'margin-left': 0,
				'margin-right': 'auto',
			};
			break;
		case 'center':
			alignCSS = {
				'margin-left': 'auto',
				'margin-right': 'auto',
			};
			break;
		case 'right':
			alignCSS = {
				'margin-right': 0,
				'margin-left': 'auto',
			};
			break;
	}
```

File: includes/blocks/separator/frontend.css.php [hunk 5]
```diff
@@ -14,6 +14,38 @@
...
 $border_size = '100%';
 
+/**
+ * Set alignment css function.
+ *
+ * @param string $align passed.
+ * @since x.x.x
+ * @return array
+ */
+function alignment_css( $align ) {
```
[the remark this answers is not in the document]
Comment (AbhijitNage123) [reply includes/blocks/separator/frontend.css.php:24]: Its specific to Separator. Not useful for any of our block. So keeping it same

File: src/blocks/separator/styling.js [hunk 6]
```diff
@@ -5,6 +5,30 @@
 import generateCSS from '@Controls/generateCSS';
 import generateCSSUnit from '@Controls/generateCSSUnit';
 import { getFallbackNumber } from '@Controls/getAttributeFallback';
+const alignmentCSS = ( align ) => {
```
[the remark this answers is not in the document]
Comment (AbhijitNage123) [reply src/blocks/separator/styling.js:8]: Its specific to Separator. Not useful for any of our block. So keeping it same

File: src/blocks/separator/styling.js (hunk 4 above)
Comment (AbhijitNage123) [reply src/blocks/separator/styling.js]: Done

**Review by adi3890 — APPROVED** [review]

**Pull request merged**